src/main/java/at/hannibal2/skyhanni/features/inventory/FannCost.kt

```diff
@@ -0,0 +1,258 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.InventoryUpdatedEvent
+import at.hannibal2.skyhanni.events.LorenzToolTipEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.NumberUtil.formatDouble
+import at.hannibal2.skyhanni.utils.NumberUtil.roundTo
+import at.hannibal2.skyhanni.utils.RegexUtils.groupOrNull
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.util.regex.Pattern
+
+@SkyHanniModule
+object FannCost {
+
+    private val config get() = SkyHanniMod.feature.inventory.fannCost
+    private val showCoins get() = config.coinsPerXP
+    private val showBits get() = config.xpPerBit
+    private var trainingMode: TrainingMode = TrainingMode.DAY_COUNT
+
+    private val patternGroup = RepoPattern.group("fann.inventory")
+
+    /**
+     * REGEX-TEST: Training Slot 1
+     * REGEX-TEST: Training Slot 2
+     * REGEX-TEST: Training Slot 3
+     */
+    private val trainingSlotInventoryPattern by patternGroup.pattern(
+        "training",
+        "Training Slot [1-3]",
+    )
+
+    /**
+     * REGEX-TEST: §aBegin Training
+     */
+    private val anvilPattern by patternGroup.pattern(
+        "anvil",
+        "§aBegin Training",
+    )
+
+    /**
+     * REGEX-TEST: Will earn a total of 1,000,000 EXP.
+     * REGEX-TEST: Will earn a total of 2 EXP
+     * REGEX-TEST: Will earn a total of 1,000 EXP
+     */
+    private val expEarnedPattern by patternGroup.pattern(
+        "exp.total",
+        "Will earn a total of (\\d{1,3}(?:,\\d{3})*(?:\\.\\d+)?|\\d+\\.?\\d*) EXP\\.?",
+    )
+
+    /**
+     * REGEX-TEST: EXP Per Day: 1,000
+     * REGEX-TEST: EXP Per Day: 1,230,000 (+3.4%)
+     * REGEX-TEST: EXP Per Day: 1,623,000 (+9.1%)
+     * REGEX-TEST: EXP Per Day: 1
+     */
+    private val dailyExpPattern by patternGroup.pattern(
+        "exp.daily",
+        "EXP Per Day: ([\\d,]+)(?: \\(\\+\\d{1,2}(\\.\\d{1,2})?%\\))?",
+    )
+
+    /**
+     * REGEX-TEST: Will take: 1d 0h 0m 0s
+     * REGEX-TEST: Will take: 3d 11h 10m 35s
+     * REGEX-TEST: Will take: 0d 0h 0m 1s
+     * REGEX-TEST: Will take: 493d 19h 49m 59s
+     */
+    private val durationPattern by patternGroup.pattern(
+        "training.duration.pattern",
+        "Will take: (?<day>\\d+)d (?<hr>\\d{1,2})h (?<min>\\d{1,2})m (?<sec>\\d{1,2})s",
```

Comment:
We do have time formatting functions already. Please just check against `Will Take: (?<time>.*)`